dynamic/provider_test.go

```diff
@@ -0,0 +1,58 @@
+package main
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	"github.com/hexops/autogold/v2"
+	pfbridge "github.com/pulumi/pulumi-terraform-bridge/pf/tfbridge"
+	pulumirpc "github.com/pulumi/pulumi/sdk/v3/proto/go"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func IntegrationTest(t *testing.T) {
+	if testing.Short() {
+		t.Skipf("Skipping integration test during -short")
+	}
+}
+
+func grpcTestServer(ctx context.Context, t *testing.T) pulumirpc.ResourceProviderServer {
+	defaultInfo, metadata, close := initialSetup()
+	t.Cleanup(func() { assert.NoError(t, close()) })
+	s, err := pfbridge.NewProviderServer(ctx, nil, defaultInfo, metadata)
+	require.NoError(t, err)
+	return s
+}
+
+func TestSchemaGeneration(t *testing.T) {
```

Comment:
I would be curious to compare the schema generated by this method vs the one generated by static TFGen for something large like AWS, not to check it in but to grasp the discrepancies. I suspected historically that there is some information we're not able to recover from the gRPC protocol possibly including

- whether a property is optional or not
- whether a proeprty is MaxItems=1 or not

This would be great to understand and document, as obviously a gRPC schema loading approach is preferable, if the limitations aren't there or can be worked around we could consider further leaning into it.

Reply:
The v6 protocol (and I think the v5 protocol) has support for both optional properties and MaxItems properties. It's definitely worth doing a direct comparison to see the difference.

Reply:
I opened an issue to track this.

Reply:
Thanks a ton! It might be a much cleaner and more robust way to do things than what's in place now.